src/ServicePulse.Host/vue/src/components/DataView.vue

```diff
@@ -1,33 +1,26 @@
-<script setup>
+<script setup lang="ts" generic="T">
 import { onMounted, onUnmounted, ref, watch } from "vue";
-import { useFetchFromServiceControl } from "../composables/serviceServiceControlUrls";
+// eslint-disable-next-line vue/prefer-import-from-vue
+import type { UnwrapRefSimple } from "@vue/reactivity";
+import { useTypedFetchFromServiceControl } from "../composables/serviceServiceControlUrls";
 import ItemsPerPage from "../components/ItemsPerPage.vue";
 import PaginationStrip from "../components/PaginationStrip.vue";
-const props = defineProps({
-  apiUrl: String,
-  itemsPerPageOptions: {
-    type: Array,
-    default: () => [20, 35, 50, 75],
-  },
-  itemsPerPage: {
-    type: Number,
-    default: 50,
-  },
-  autoRefreshSeconds: {
-    type: Number,
-  },
-  showPagination: {
-    type: Boolean,
-    default: true,
-  },
-  showItemsPerPage: {
-    type: Boolean,
-    default: false,
-  },
-});
 
-const refreshTimer = ref();
-const items = ref([]);
+const props = withDefaults(
+  defineProps<{
+    itemsType: T[];
+    apiUrl: string;
+    itemsPerPageOptions?: number[];
+    itemsPerPage?: number;
+    autoRefreshSeconds: number;
+    showPagination?: boolean;
+    showItemsPerPage?: boolean;
+  }>(),
+  { itemsPerPageOptions: () => [20, 35, 50, 75], itemsPerPage: 50, showPagination: true, showItemsPerPage: false }
+);
+
+const refreshTimer = ref<number>();
```

Comment:
Does this default to 0? Is it OK to call `clearInterval` on a 0?

Reply:
Also, I wrote this as a `ref` but I'm wondering if it should be a `let`. We don't bind this to anything.

Reply:
const is fine @mikeminutillo

Reply:
it defaults to `undefined`

Reply:
I meant should it not be a `ref` at all. i.e.

```js
let refreshTimer : number;
```

Reply:
fixed